app/helpers/samples_helper.rb

```diff
@@ -165,7 +165,11 @@ def parsed_samples_for_s3_path(s3_path, project_id, host_genome_id)
     s3_prefix = parsed_uri.path.sub(%r{^/(.*?)/?$}, '\1/')
 
     begin
-      entries = S3_CLIENT_LOCAL.list_objects_v2(bucket: s3_bucket_name, prefix: s3_prefix).contents.map(&:key)
+      entries = []
+      S3_CLIENT_LOCAL.list_objects_v2(bucket: s3_bucket_name, prefix: s3_prefix).each do |response|
+        entries += response.contents.map(&:key)
+        break if entries.length >= 10_000
```

Comment:
I think we don't bubble these errors back up, but for on call debugging assistance could you do something like `if entries.length >= 10_000` then `Rails.logger.info("User tried to list more than 10,000 objects in #{s3_path}")`